PocketKit/Sources/Sync/PocketSource.swift

```diff
@@ -649,15 +649,17 @@ extension PocketSource {
     }
 
     public func fetchOrCreateSavedItem(with remoteID: String, and remoteParts: SavedItem.RemoteSavedItem?) -> SavedItem? {
+        let savedItem = (try? space.fetchSavedItem(byRemoteID: remoteID))
+
         guard let remoteParts, let url = URL(string: remoteParts.url) else {
-            Log.breadcrumb(category: "sync", level: .warning, message: "Skipping updating of SavedItem because we do not have a valid url or we have no remoteParts")
-            return nil
+            Log.breadcrumb(category: "sync", level: .debug, message: "SavedItem found and don't need to create one")
+            return savedItem
         }
 
-        let savedItem = (try? space.fetchSavedItem(byRemoteID: remoteID)) ?? SavedItem(context: space.context, url: url, remoteID: remoteID)
-        savedItem.update(from: remoteParts, with: space)
+        let remoteSavedItem = SavedItem(context: space.context, url: url, remoteID: remoteID)
```

Comment:
Don't we still want to update the SavedItem object we pulled from CoreData with the latest data?

Reply:
This function was meant to create a savedItem based on the user interaction with the card, but let me think about this

Reply:
I updated the savedItem with remote parts! Rather than enforcing that there is a url, we assume that url is already associated with the savedItem as we now have it as non nil in core data.